File: AkkaWithZeroMQ/src/main/java/org/akka/essentials/zeromq/example2/MyActorSystem.java
package org.akka.essentials.zeromq.example2;

import akka.actor.ActorSystem;
import akka.actor.Props;

public class MyActorSystem {

	/**
	 * @param args
	 */
	public static void main(String[] args) {
		ActorSystem system = ActorSystem.create("zeromqTest");
		system.actorOf(new Props(WorkerTaskA.class), "workerA");
		system.actorOf(new Props(WorkerTaskB.class), "workerB");
		system.actorOf(new Props(RouterActor.class), "router");
	}

}

File: AkkaWithZeroMQ/src/main/scala/org/akka/essentials/zeromq/example2/MyActorSystem.scala
package org.akka.essentials.zeromq.example2
import akka.dispatch.Await
import akka.actor.ActorSystem
import akka.util.Timeout
import akka.actor.Props

object MyActorSystem {

  def main(args: Array[String]): Unit = {
    val system = ActorSystem("zeromqTest")
    system.actorOf(Props[RouterActor], name = "router")
    system.actorOf(Props[WorkerTaskA], name = "workerA")
    system.actorOf(Props[WorkerTaskB], name = "workerB")
  }
}